src/Website/Tool/PimcoreUrl.php

```diff
@@ -30,7 +30,8 @@ public function __invoke(array $urlOptions = [], $name = null, $reset = false, $
     {
         // merge all parameters from request to parameters
         if (!$reset && $this->requestHelper->hasMainRequest()) {
-            $urlOptions = array_replace($this->requestHelper->getMainRequest()->attributes->get('_route_params'), $urlOptions);
+            $params = $this->requestHelper->getMainRequest()->attributes->get('_route_params') ?? [];
+            $urlOptions = array_replace($params, $urlOptions);
```

Comment:
```suggestion
            $urlOptions = array_replace($this->requestHelper->getMainRequest()->attributes->get('_route_params') ?? [], $urlOptions);
```
Thank you for your suggestion, just wondering, why isn't like this instead? for readability?

Reply:
there is always room for improvement ;-)

Reply:
ahh ok, was just curious and wondering if maybe i've missed something 😄 
`$params` could be used to check if is empty to avoid performing an `array_replace` where the base is just empty array. eg.
```suggestion
            $params = $this->requestHelper->getMainRequest()->attributes->get('_route_params') ?? [];
            if (!empty($params)){
                 $urlOptions = array_replace($params, $urlOptions);
            }
```

Reply:
or even better 😃 

```suggestion
            $urlOptions = array_replace($this->requestHelper->getMainRequest()->attributes->get('_route_params', []), $urlOptions);
```

the second parameter of `get()` is for default value fallback
https://github.com/symfony/symfony/blob/06e630c175121837828e806493a285d6bd363dd6/src/Symfony/Component/HttpFoundation/ParameterBag.php#L81